tests/umock_c_with_lock_factory_int/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+cmake_minimum_required(VERSION 2.8.11)
+
+set(theseTestsName umock_c_with_lock_factory_int)
+
+set(${theseTestsName}_test_files
+umock_c_with_lock_factory_int.c
+)
+
+if(WIN32)
+# For Windows use CreateThread et all
+set(test_platform_c_files
+./minipal/umock_threadapi_win32.c
+../../src/umock_lock_factory_windows.c
+)
+else()
+# For others assume pthreads (this should be a real check for pthreads)
+set(test_platform_c_files
+./minipal/umock_threadapi_pthread.c
+../../src/umock_lock_factory_pthread.c
+)
+endif()
+
+set(${theseTestsName}_c_files
+../../src/umock_c.c
```

Comment:
any reason not to just use the lib?

Reply:
There is another task by which I'll do that. I do not recall the exact reason it got coded like this in the first place.

---
In reply to: [530658693](https://github.com/Azure/umock-c/pull/177#discussion_r530658693) [](ancestors = 530658693)